### ISSUE TITLE
Better handling of crashing processes

### ISSUE THREAD

**Issue by ddollar** [convox/rack#68]
If a process crashes immediately on boot and is not fronted by an ELB it will thrash forever trying to restart.

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;nTDS8Nu4&#x2F;175-better-handling-of-crashing-processes"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> better handling of crashing processes</a>

**Comment by nzoschke** [comment]
This is tricky since the deploy and CloudFormation update can finish, but ECS will have problems behind the scenes. Some ways to improve the situation:

* Guarantee that logging happens during process churning, ensuring app log visibility
* Send notifications about ECS steady state
* Send notifications about ECS unhappy state
* Implement `convox debug` or similar that exposes ECS events and task errors

**Comment by nzoschke** [comment]
The notifications I'd like to see next are very app centric:

* Rolling deploy of `httpd` completed
* Rolling deploy of `httpd` failed by not completing in 15 minutes
* Process `worker` on app `httpd` is crashing

There could be some improvements to the rack notifications too:

* Rack update to `X` completed
* Rack update to `X` rolled back
* Rack instance i-123435 terminated due to a bad docker daemon

**Comment by nzoschke** [comment]
Aside from the instance termination stuff, the ECS APIs have all the events, its a matter of improving the strategy for polling events and synthesizing them into more specific messages.

The current `Rack in steady state` or `Detected capacity warning` notifications are fairly naive.

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;myJu43RK&#x2F;271-check-out-console-rack-spamming-slack-w-ready-state-message"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Check out console&#x2F;rack spamming slack w&#x2F; ready state message</a>

**Comment by nzoschke** [comment]
I feel pretty good about logging during process churning, though I haven't tested it recently.

There are some related issues here:

* https://github.com/convox/rack/issues/182
* https://github.com/convox/rack/issues/441

All of these depend on understanding the state changes of the cluster well.

**Comment by contra** [comment]
This would be **really** **really** appreciated.

Had an issue where the dockerfile worked fine locally, but was broken on deploy (didn't copy the bin file used to start the process). Process thrashed infinitely and the application was unavailable with no notification that something was wrong until newrelic emailed me that the health endpoint was offline (wat) and I manually rolled back to the last release.

Even though the process was infinitely restarting and none of the ports I configured were opened, convox still thought it was healthy enough to route traffic to. The rolling updates documentation states the opposite of this (https://convox.com/docs/rolling-updates/).

<img width="788" alt="screen shot 2016-07-20 at 2 11 58 pm" src="https://cloud.githubusercontent.com/assets/425716/16997697/8d14625a-4e84-11e6-956a-97521c2b458d.png">
